docs/core/extensions/logger-message-generator.md

```diff
@@ -111,11 +111,12 @@ Consider the example logging output when using the `JsonConsole` formatter.
 
 When using the `LoggerMessageAttribute` on logging methods, there are some constraints that must be followed:
 
-- Logging methods must be `static`, `partial`, and return `void`.
+- Logging methods must be `partial` and return `void`.
 - Logging method names must *not* start with an underscore.
 - Parameter names of logging methods must *not* start with an underscore.
 - Logging methods may *not* be defined in a nested type.
 - Logging methods *cannot* be generic.
+- If a logging method is `static`, the `ILogger` instance is required as a parameter.
```

Comment:
Doc comments on the attribute just mentions `ILogger` must be a parameter. It doesn't say that it's applicable only to static methods

https://github.com/dotnet/runtime/blob/2330b4bef645bf8bcd3011f35968e55192c697f5/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerMessageAttribute.cs#L16

Is the doc comment outdated? or is it a general requirement regardless of the static-ness of the method?

Reply:
The doc comment also mentions a `LogLevel` parameter as a requirement.

Reply:
Reading through the code. It seems the doc comment is outdated or incorrect.

https://github.com/dotnet/runtime/blob/2330b4bef645bf8bcd3011f35968e55192c697f5/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs#L403-L411